### PR TITLE
Document newton more.

### PR DESCRIPTION
**Pull request by Hixie**
This started off as me just wanting to add periods to the end of the comments in this library.

It got a bit out of control.

cc @chinmaygarde who probably will need to carefully check some of these comments for accuracy.

### CODE REVIEW

File: packages/newton/test/all.dart [hunk 1]
```diff
@@ -1,9 +0,0 @@
-// Copyright (c) 2016 The Chromium Authors. All rights reserved.
-// Use of this source code is governed by a BSD-style license that can be
-// found in the LICENSE file.
-
-import 'newton_test.dart' as newton_test;
-
-void main() {
-  newton_test.main();
```
Comment (abarth) [inline packages/newton/test/all.dart]: Isn't the "all.dart" file needed by the bots?

Comment (Hixie) [inline packages/newton/test/all.dart]: yeah, I misunderstood what was going on here. Resurrected it.

**Comment by chinmaygarde** [comment]
lgtm.

All simple simulations are stateless. Using simulation groups is where it gets tricky since you need the final state of one simulation to configure the next. I remember @HansMuller once had the suggestion that maybe `SimulationGroup` does not belong in this library. We could move that out and then be able to make guarantees about the statelessness of the simulations (simple to use and do only one thing well). Could make the library easier to reason about and use.

**Comment by Hixie** [comment]
I dunno. Any non-analytical solution will have that problem. e.g. a Brownian motion simulation. Also, it's not like we ever need to go back in time in practice with simulations. We have Curves and Tweens for that case.

**Pull request merged**